files/image_config/hostcfgd/hostcfgd

```diff
@@ -219,6 +220,474 @@ class AaaCfg(object):
+            cmd = 'ip link delete mgmt'
+            if run_command(cmd) == True:
+                self.mgmt_vrf_table_created = False
+
+    def configure_mgmt_vrf(self):
+        self.create_mgmt_vrf_table()
+
+        cmd = 'ip link set dev mgmt up'
+        if run_command(cmd) == False:
+            return
+
+        cmd = 'ip link add lo-m type dummy'
+        if run_command(cmd) == False:
+            return
+
+        cmd = 'ip addr add 127.0.0.1/8 dev lo-m'
+        if run_command(cmd) == False:
+            return
+
+        cmd = 'ip link set lo-m up'
+        if run_command(cmd) == False:
+            return
+
+        cmd = 'ip link set dev lo-m master mgmt'
+        if run_command(cmd) == False:
+            return
+
+        self.bind_mgmt_intf()
+
+    def bind_mgmt_intf(self):
+        keys = self.config_db.get_keys(CFG_MGMT_INTF_TABLE)
+
+        mgmt_intf_bound = False
+        for it in keys:
+            if type(it) is unicode:
+                mgmt_intf_name = it
+
+                cmd = 'ip link set dev {} master mgmt'.format(mgmt_intf_name)
+                if run_command(cmd) == False:
+                    return
+
+                mgmt_intf_bound = True
+                break
+        return mgmt_intf_bound
+
+    def mgmt_vrf_add_default_routes(self):
+        keys = self.config_db.get_keys(CFG_MGMT_INTF_TABLE)
+        for it in keys:
+            if type(it) is unicode:
+                continue
+
+            key = it[0] + "|" + it[1]
+
+            entry = self.config_db.get_entry(CFG_MGMT_INTF_TABLE, key)
+            if 'gwaddr' in entry:
+                mgmt_intf_gwaddr = entry['gwaddr']
+                self.handle_mgmt_intf_gwaddr_cfg("ADD", it[0], mgmt_intf_gwaddr)
+
+    def unconfigure_mgmt_vrf(self):
+        cmd = 'ip link delete lo-m'
+        if run_command(cmd) == True:
+            self.delete_mgmt_vrf_table()
+
+        return
+
+    def handle_mgmt_intf_gwaddr_cfg(self, op, ifname, gwaddr):
+        rc = False
+        keys = self.config_db.get_keys(CFG_MGMT_VRF)
+        if keys:
+            cmd = 'ip route del table {} 0.0.0.0/0'.format(MGMT_VRF_TABLE_ID)
+            run_command(cmd)
+            
+            if op == "ADD":
+                cmd = 'ip route add table {} 0.0.0.0/0 via {}'.format(MGMT_VRF_TABLE_ID, gwaddr)
+                run_command(cmd)
+            rc = True
+
+        return rc
+
+    def handle_mgmt_intf_creation(self, mgmt_intf_key):
+        keys = self.config_db.get_keys(CFG_MGMT_VRF)
+        if keys:
+            if self.bind_mgmt_intf() == True:
+                self.mgmt_vrf_add_default_routes()
+
+    def handle_mgmt_vrf_cfg(self, key, data):
+        try:
+            if data:
+                if data['mgmtVrfEnabled'] == 'true':
+                    self.configure_mgmt_vrf()
+                else:
+                    self.unconfigure_mgmt_vrf()
+            else:
+                self.unconfigure_mgmt_vrf()
+        except Exception as inst:
```

Comment:
what are the possible reasons for this exception? If mvrf enable/disable fails, will eth0 continue to remain in default/mgmt vrf and work normally in default/mgmt vrf?